crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -1340,8 +1331,11 @@ impl<F: Clone> UpdateTracker<F, PaymentData<F>, api::PaymentsRequest> for Paymen
                         client_source,
                         client_version,
                         customer_acceptance: payment_data.payment_attempt.customer_acceptance,
-                        shipping_cost,
-                        order_tax_amount,
+                        shipping_cost: payment_data.payment_intent.shipping_cost,
```

Comment:
we need not have these fields in the flattened format in update right? can we use the amount struct in update as well?

Reply:
Based on the flow, we would need to restrict the updating of fields. Let's say in payments update flow, we don't want to update surcharge and tax amount.